controllers/complianceeventsapi/server.go

```diff
@@ -802,6 +795,204 @@ func postComplianceEvent(db *sql.DB, w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func getComplianceEventsQuery(whereClause string, queryArgs *queryOptions) string {
+	// Example query:
+	//   SELECT compliance_events.id, compliance_events.compliance, ...
+	//     FROM compliance_events
+	//   LEFT JOIN clusters ON compliance_events.cluster_id = clusters.id
+	//   LEFT JOIN parent_policies ON compliance_events.parent_policy_id = parent_policies.id
+	//   LEFT JOIN policies ON compliance_events.policy_id = policies.id
+	//   WHERE (policies.name=$1 OR policies.name=$2) AND (policies.kind=$3)
+	//   ORDER BY compliance_events.timestamp desc
+	//   LIMIT 20
+	//   OFFSET 0 ROWS;
+	return fmt.Sprintf(`%s%s
+	ORDER BY %s %s
+	LIMIT %d
+	OFFSET %d ROWS;`,
+		generateGetComplianceEventsQuery(queryArgs.IncludeSpec),
+		whereClause,
+		strings.Join(queryArgs.Sort, ", "),
+		queryArgs.Direction,
+		queryArgs.PerPage,
+		(queryArgs.Page-1)*queryArgs.PerPage,
```

Comment:
I think these will need adjustments for the CSV, because it shouldn't do any pagination for that. I don't know if skipping the LIMIT and OFFSET would be right in all cases, maybe it should still limit it to some number (just so it's not an absolutely massive amount of data coming from the database all at once), but be sure to go through all of the pages.

Reply:
page can be omitted by users if they want. but some users maybe need csv file just like `GET` endpoint result?!!  personally, i like this because we give more options to users. I don't know what @mprahl opinion. We can discuss this!

Reply:
@yiraeChristineKim, it's a good point. I think we can accept pagination arguments but default `per_page` to unlimited if not explicitly set by the user for CSV reports where as the list endpoint will still require a maximum `per_page`.